### PR TITLE
feat(default-theme): order details component with ability to pay for an order

### LINKED ISSUE #664: [FEATURE] Add order endpoint to API client and implement view
### Description

success page

### Acceptance criteria

- [x] add endpoint from #262 (added via https://github.com/DivanteLtd/shopware-pwa/pull/752)
- [x] implement view using provided endpoint

### LINKED ISSUE #717: [FEATURE] Implement order details component
### Description

related #664 

### Acceptance criteria

- [x] implement order details component based on SFUI designs
- [x] component is used in my account and thank you page
- [x] in section status there is a payment button showing when payment is available (this is implemented, only place it inside order details)

### PR DESCRIPTION
**Pull request by mkucmus**
## Changes

closes #717 
closes #664 

### Checklist

- [x] I followed [contributing](https://github.com/DivanteLtd/shopware-pwa/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: packages/helpers/src/order/getOrderShippingMethodId.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * get shipping method's id from open transaction
+ *
+ * @alpha
+ */
+export function getOrderShippingMethodId(order: Order): string | undefined {
+  const openDelivery = order?.deliveries?.find(
+    ({ stateMachineState }) =>
+      stateMachineState && stateMachineState.technicalName === "open"
```
Comment (patzick) [inline packages/helpers/src/order/getOrderShippingMethodId.ts:11]:
```suggestion
      stateMachineState?.technicalName === "open"
```

File: packages/helpers/src/order/getOrderPaymentMethodId.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * get payment method's id from open transaction
+ *
+ * @alpha
+ */
+export function getOrderPaymentMethodId(order: Order): string | undefined {
+  const openTransaction = order?.transactions?.find(
+    ({ stateMachineState }) =>
+      stateMachineState && stateMachineState.technicalName === "open"
```
Comment (patzick) [inline packages/helpers/src/order/getOrderPaymentMethodId.ts:11]:
```suggestion
      stateMachineState?.technicalName === "open"
```

File: packages/default-theme/pages/_lang/success-page.vue [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
-    </a>
     <SfHeading
       title="Thank you"
       subtitle="for shopping with us!"
       class="success-page__heading"
     />
-    <SfButton @click="$router.push($i18n.path('/'))">
+    <SfDivider />
+    <SwOrderDetails :order-id="orderId" />
+    <SfButton @click="$router.push('/')">
```
Comment (patzick) [inline packages/default-theme/pages/_lang/success-page.vue:10]:
```suggestion
    <SfButton @click="$router.push($i18n.path('/'))">
```

File: packages/default-theme/components/SwPersonalDetails.vue [hunk 4]
```diff
@@ -0,0 +1,58 @@
+<template>
+  <div class="sw-personal-details">
+    <div class="sw-personal-details__content">
+      <h4 class="sw-personal-details__title">Personal details</h4>
+      <p class="content">{{ firstName }} {{ lastName }}<br /></p>
+      <p class="content">
+        {{ email }}
+      </p>
```
Comment (patzick) [inline packages/default-theme/components/SwPersonalDetails.vue:8]: add slot here and use this in `PersonalDetailsSummary` :)

File: packages/default-theme/components/SwGenericMethod.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    <p class="sw-generic-method__content">
+      {{ method.name }}
+    </p>
+  </div>
+</template>
+
+<script>
+import { SfInput, SfButton, SfAlert, SfTable } from '@storefront-ui/vue'
+
+import SwPluginSlot from 'sw-plugins/SwPluginSlot'
```
Comment (patzick) [inline packages/default-theme/components/SwGenericMethod.vue:13]: these imports are not used in component

also, please rename to `SwCheckoutMethod`

File: packages/default-theme/components/SwAddress.vue [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      <p class="content" v-if="phoneNumber">
+        {{ phoneNumber }}
+      </p>
+    </div>
+  </div>
+</template>
+
+<script>
+import { SfInput, SfButton, SfAlert, SfTable } from '@storefront-ui/vue'
+import SwPluginSlot from 'sw-plugins/SwPluginSlot'
```
Comment (patzick) [inline packages/default-theme/components/SwAddress.vue:18]: imports to cleanup

File: packages/default-theme/components/SwAddress.vue [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    <div class="sw-address__content">
+      <h4 class="sw-address__title">{{ addressTitle }}</h4>
+      <p class="content">
+        {{ street }} {{ apartment }}, {{ zipcode }}<br />
+        {{ city }}
+      </p>
+      <p class="content" v-if="phoneNumber">
+        {{ phoneNumber }}
+      </p>
+    </div>
```
Comment (patzick) [inline packages/default-theme/components/SwAddress.vue:12]: please reuse this component in `ShippingAddressSummary`

File: packages/default-theme/cms/elements/CmsElementCategorySidebarFilter.vue [hunk 8]
```diff
@@ -196,7 +196,7 @@ export default {
       return true
     },
     isGridView() {
-      return this.$store.state.isGridView
+      return this.$store.state.isGridView || false
```
Comment (patzick) [inline packages/default-theme/cms/elements/CmsElementCategorySidebarFilter.vue:199]:
```suggestion
      return !!this.$store.state.isGridView
```

**Comment by akrajna** [comment]
Order details in My account:
* the numbers in quantity column should be as big as the price and and be centered
* the currency should be in one row with the price
* font size for all the prices should be the same (Price, Amount, Subtotal, Shipping Total price)
Order details

Thank you page:
* the numbers in quantity column should be as big as the price
* the price in Amount column should be right aligned
* only Subtotal, Shipping and Total price  should be in bold, other prices should not
* font size for all the prices should be the same (Price, Amount, Subtotal, Shipping Total price)
thank you page

**Comment by akrajna** [comment]
On mobile 'Thank you page' and 'order details in My account'  the order details needs to be adjusted:
* Item and price is mixed and Quantity and amount are mixed it my be confusing for the user
MobileThankYou
consider figma design  as below:
figmaOrderDetails

**Comment by akrajna** [comment]
as some of the comments concern UI, a new issue has been created #801

**Pull request merged**